src/main/java/in/twizmwaz/cardinal/module/modules/enderpearlReset/EnderpearlReset.java

```diff
@@ -0,0 +1,51 @@
+package in.twizmwaz.cardinal.module.modules.enderpearlReset;
+
+import in.twizmwaz.cardinal.event.KitApplyEvent;
+import in.twizmwaz.cardinal.module.Module;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.entity.ProjectileLaunchEvent;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.UUID;
+
+public class EnderpearlReset implements Module {
+
+    private Map<UUID, Entity> thrownPearls = new HashMap<>();
+
+    @Override
+    public void unload() {
+        HandlerList.unregisterAll(this);
+    }
+
+    @EventHandler
+    public void onProjectileLaunch(ProjectileLaunchEvent event) {
+        if (event.getEntity().getType().equals(EntityType.ENDER_PEARL)) {
+            if (event.getEntity().getShooter() instanceof Player) {
+                Player shooter = (Player) event.getEntity().getShooter();
+                thrownPearls.put(shooter.getUniqueId(), event.getEntity());
+            }
+        }
+    }
+
+    @EventHandler
+    public void onApplyingKit(KitApplyEvent event) {
+        if (event.getKit().isResetPearls()) {
+            Iterator<Map.Entry<UUID, Entity>> iterator = thrownPearls.entrySet().iterator();
```

Comment:
I would replace this style of a iterator with something that is more conventional, something along the lines of this:

```java
for (Map.Entry<UUID, Entity> entry : thrownPearls.entrySet()) {
    if (entry.getKey().equals(event.getPlayer().getUniqueId())) {
        if (!entry.getValue().isDead()) {
            entry.getValue().remove();
        }
    }
}
```
}